### PR TITLE
Feature/test2

### PR DESCRIPTION
**Pull request by RoshanShrestha-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
- [ ] The scripts were tested successfully and it produced satisfactory results 

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**
Yes
- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**
Yes
- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
@RoshanShrestha-NOAA  -- I have picked up your PR and will begin testing it. 
Can you please sync your fork w/ EVS develop? I think there have been 1 or 2 merges into EVS develop since you submitted this PR and I want your feature/test2 to have the latest code.

**Comment by RoshanShrestha-NOAA** [comment]
Synced my forks. Thank you.

**Comment by ShelleyMelchior-NOAA** [comment]
I have run the nam and rap g2o stats. Please review the output.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale`
log files:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_nam_grid2obs_stats_00.o66097934`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_rap_grid2obs_stats_00.o66124995`

Also take a look at your COMOUT structure. I think the path needs to be tweaked. You have:
`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/nam.20230622` That one is OK. But,
`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/atmos.20230622/nam/grid2obs/nam.20230622` has an extra dir (`nam.20230622`) that shouldn't be there. This path should end at `grid2obs/`. The same is true for rap.

You can also update your ecf scripts to remove the module loads for `proj`, `geos`, and `netcdf`. Those are inherently loaded w/ the load of ve/evs/1.0. See below:
`$ module show ve/evs/1.0`
`---------------------------------------------------------------------------------------------------`
`   /apps/dev/modulefiles/ve/evs/1.0:`
`---------------------------------------------------------------------------------------------------`
`whatis("EVS ")`
`load("intel","PrgEnv-intel","craype","hdf5","netcdf4","geos","proj")`
`setenv("VIRTUAL_ENV","/apps/dev/ve/evs/1.0")`
`setenv("PYTHONPATH","/apps/dev/ve/evs/1.0/lib/python3.10/site-packages")`
`prepend_path("PATH","/apps/dev/ve/evs/1.0/bin")`
`help([[EVS`
`]])`

File: ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf]: Can remove this module load. `netcdf` is inherently loaded with ve/evs/1.0.

File: ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf]: Can remove this module load. `proj` is inherently loaded w/ ve/evs/1.0.

File: ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf]: Can remove this module load. `geos` is inherently loaded w/ ve/evs/1.0.

File: ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf]: Can remove this module load. `netcdf` is inherently loaded w/ ve/evs/1.0.

File: ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf]: Can remove this module load. `proj` is inherently loaded w/ ve/evs/1.0.

File: ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf]: Can remove this module load. `geos` is inherently loaded w/ ve/evs/1.0.

**Review by ShelleyMelchior-NOAA — CHANGES_REQUESTED** [review]

File: ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf
Comment (RoshanShrestha-NOAA) [reply ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf]: Done

File: ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf
Comment (RoshanShrestha-NOAA) [reply ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf]: Done

File: ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf
Comment (RoshanShrestha-NOAA) [reply ecf/scripts/mesoscale/stats/jevs_nam_grid2obs_stats.ecf]: Done

File: ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf
Comment (RoshanShrestha-NOAA) [reply ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf]: Removed proj

File: ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf
Comment (RoshanShrestha-NOAA) [reply ecf/scripts/mesoscale/stats/jevs_rap_grid2obs_stats.ecf]: Removed geos

**Comment by RoshanShrestha-NOAA** [comment]
> `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/atmos.20230622/nam/grid2obs/nam.20230622` has an extra dir (`nam.20230622`) that shouldn't be there. This path should end at `grid2obs/`. The same is true for rap.

The extra dir nam.20230622 are removed.

**Comment by ShelleyMelchior-NOAA** [comment]
Great! I will re-run with these changes applied just to make sure COMOUTsmall works correctly. Once confirmed, go ahead and remove the commented out lines in
`scripts/mesoscale/stats/exevs_nam_grid2obs_stats.sh` (341 and 344)
and
`scripts/mesoscale/stats/exevs_rap_grid2obs_stats.sh` (344 and 347).

How do the output stats files look, @RoshanShrestha-NOAA ?

**Comment by RoshanShrestha-NOAA** [comment]
I have checked the stat outputs. All looks normal. COMOUTsmall looks working correctly as well.

**Comment by ShelleyMelchior-NOAA** [comment]
Great, @RoshanShrestha-NOAA ! I will do a final code review and merge into the auth EVS develop.

**Comment by ShelleyMelchior-NOAA** [comment]
@RoshanShrestha-NOAA -- I only tested the stats in this PR. However, you included some plotting code.
@MarcelCaron-NOAA  -- what is the plan here? Should I include the plotting code in this PR knowing your plotting PR will subplant these files? Or should I work w/ Roshan to have him remove the plotting scripts from this PR?

**Comment by PerryShafran-NOAA** [comment]
Shelley - Marcel's plots are ONLY for precip and snowfall.  I would guess that Roshan's plotting is just for grid2obs/grid2grid stuff.  That sounds like the division of labor for the mesoscale processes.

**Comment by ShelleyMelchior-NOAA** [comment]
Sorry, let me be more clear. @MarcelCaron-NOAA has a forthcoming PR that will cover grid2obs plots for mesoscale for all sfc and upper air metrics.

**Comment by PerryShafran-NOAA** [comment]
Thanks for that clarification!  I am not sure I was aware of that.  It's good for Roshan to be clear what he wants to be tested with this PR.

Perry

**Comment by MarcelCaron-NOAA** [comment]
@PerryShafran-NOAA and @ShelleyMelchior-NOAA there was some confusion a few weeks ago as to who is handling the grid2obs plotting portion for the mesoscale component of EVS, so I think that’s why some plotting scripts are included in, but not necessary for, this PR.  Is that correct @RoshanShrestha-NOAA?

The grid2grid plotting PR is submitted and the grid2obs plotting PR is forthcoming, so this is when the mesoscale plotting capability should be tested.  As for the mesoscale plotting scripts that exist in this PR, I can either update the versions in my local feature branch after this PR is approved and when I merge the changes made to dev, or the scripts can be deleted here.

**Comment by RoshanShrestha-NOAA** [comment]
Hi @MarcelCaron-NOAA, you are right. Let me know if I can do anything to clarify this confusion. Thank you.

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @RoshanShrestha-NOAA  and @MarcelCaron-NOAA  -- I recommend this course of action as the least messy for the situation.

Roshan, please create a new feature branch that is a subset of feature/test2, to only include the stats files that are needed in feature/test2. Let's omit the plots code. Then submit a new PR  from the new feature branch to cover just the stats. Once that PR is submitted, I will close this current PR w/o a merge, and instead will merge the new PR. All of the g2o plots code will be included in Marcel's forthcoming PR.

How does this sound? Let me know if you have any questions.

**Comment by RoshanShrestha-NOAA** [comment]
Hi @ShelleyMelchior-NOAA, OK, I will do so. Thank you.

**Comment by RoshanShrestha-NOAA** [comment]
Hi @ShelleyMelchior-NOAA, A new PR submitted following your suggestions. The plotting scripts are removed in that new feature branch. Thank you.

**Comment by ShelleyMelchior-NOAA** [comment]
This PR is related to PR #144

**Comment by ShelleyMelchior-NOAA** [comment]
Closing this PR as PR #144 has been merged into auth EVS develop.

**Pull request closed without merging**
